pkg/sdscomm/sdscomm.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sdscomm
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/trustbloc/edge-core/pkg/log"
+	sdsclient "github.com/trustbloc/edv/pkg/client"
+	"github.com/trustbloc/edv/pkg/restapi/messages"
+	"github.com/trustbloc/edv/pkg/restapi/models"
+)
+
+var logger = log.New("edge-agent-sdscomm")
+
+type SDSComm struct {
+	sdsServerURL  string
+	agentUsername string
+	sdsClient     *sdsclient.Client
+}
+
+type DIDData struct {
+	DID      json.RawMessage `json:"did,omitempty"`
+	SignType string          `json:"signType,omitempty"`
+	Name     string          `json:"name,omitempty"`
+}
+
+func New(sdsServerURL, agentUsername string) *SDSComm {
+	return &SDSComm{
+		sdsServerURL:  sdsServerURL,
+		agentUsername: agentUsername,
+		sdsClient:     sdsclient.New(sdsServerURL),
+	}
+}
+
+// CreateDIDVault creates the user's DID vault if it doesn't exist.
+func (e *SDSComm) CreateDIDVault() error {
+	didVaultID := e.GetDIDVaultID()
+
+	_, err := e.sdsClient.CreateDataVault(&models.DataVaultConfiguration{ReferenceID: didVaultID})
```

Comment:
why are we ignoring the vault id?

Reply:
if we got no error, it means a new vault was created... its ID needs to be pushed to hub-auth

Reply:
this API is weird... you pass in the ID.. and I suppose it returns it back.. ?

Reply:
Ah... I see why this is confusing. Let me try to explain...

1. The vault ID is passed in to the SDS client so that it can create a vault with that specified ID.
2. The SDS server gives back the location (which is a full URL) of the newly created data vault (per the spec). This will look something like `https://localhost:8080/encrypted-data-vaults/newDIDVaultID` for our TrustBloc SDS implementation.
3. In this PR, I haven't added the ability to push back any data to hub-auth, so right now that location is useless to me. In a future PR, I will indeed need it, and it'll need to be pushed to the bootstrap data in hub-auth.

Hope that makes sense!